packages/next-swc/crates/next-core/js/src/entry/app/edge-page-bootstrap.ts

```diff
@@ -0,0 +1,103 @@
+import 'next/dist/esm/server/web/globals'
+import { adapter } from 'next/dist/server/web/adapter'
+import { RSC_VARY_HEADER } from 'next/dist/client/components/app-router-headers'
+import { IncrementalCache } from 'next/dist/server/lib/incremental-cache'
+import { renderToHTMLOrFlight } from 'next/dist/server/app-render/app-render'
+;('TURBOPACK { chunking-type: isolatedParallel }')
+import entry from 'APP_ENTRY'
+import BOOTSTRAP from 'APP_BOOTSTRAP'
+import { createManifests, installRequireAndChunkLoad } from './manifest'
+import type { NextRequest, NextFetchEvent } from 'next/server'
+import type { RenderOpts } from 'next/dist/server/app-render/types'
+import type { ParsedUrlQuery } from 'querystring'
+
+installRequireAndChunkLoad()
+
+Error.stackTraceLimit = 100
+
+const { clientReferenceManifest } = createManifests()
+
+const MIME_TEXT_HTML_UTF8 = 'text/html; charset=utf-8'
+
+async function render(request: NextRequest, event: NextFetchEvent) {
+  const renderOpt: Omit<
+    RenderOpts,
+    'App' | 'Document' | 'Component' | 'pathname'
+  > & { params: ParsedUrlQuery } = {
+    // TODO params
+    params: {},
+    supportsDynamicHTML: true,
+    dev: true,
+    buildId: 'development',
+    buildManifest: {
+      polyfillFiles: [],
+      rootMainFiles: BOOTSTRAP.filter((path) => path.endsWith('.js')),
+      devFiles: [],
+      ampDevFiles: [],
+      lowPriorityFiles: [],
+      pages: {
+        '/_app': [],
+      },
+      ampFirstPages: [],
+    },
+    ComponentMod: {
+      ...entry,
+      __next_app_require__: __next_require__,
+      pages: ['page.js'],
+    },
+    clientReferenceManifest,
+    runtime: 'nodejs',
+    serverComponents: true,
+    assetPrefix: '',
+    pageConfig: {},
+    reactLoadableManifest: {},
+    // TODO nextConfigOutput
+    nextConfigOutput: undefined,
+  }
+
+  const tranform = new TransformStream()
+  const response = new Response(tranform.readable)
+
+  let [, pathname, query] = /^([^?]*)(.*)$/.exec(request.url)!
```

Comment:
Nit: the following would be clearer.

```suggestion
  let { pathname, search: query } = new URL(request.url, "next://")
```